GPU enabled unit-tests fail to compile

## Environment info
1. Operating System: Ubuntu 16.04, CUDA v8.0
2. Compiler: gcc v5.4.0
3. Package used (python/R/jvm/C++): C++ unit-testing with GPU compilation enabled.
4. `xgboost` version used: built from source
5. Commit hash: e8a659795763

## Steps to reproduce
Set to the above commit hash (currently the latest commit)
```bash
mkdir build
cd build
cmake .. -DUSE_CUDA=ON -DUSE_NCCL=ON -DGOOGLE_TEST=ON
make
```

## Error seen
Showing the just right output.
```
[ 76%] Building NVCC (Device) object CMakeFiles/cuda_compile.dir/tests/cpp/predictor/cuda_compile_generated_test_gpu_predictor.cu.o
include/xgboost/base.h(167): error: no operator "<<" matches these operands
            operand types are: std::ostream << float

1 error detected in the compilation of "/tmp/tmpxft_000037ae_00000000-15_test_gpu_predictor.compute_61.cpp1.ii".
CMake Error at cuda_compile_generated_test_gpu_predictor.cu.o.cmake:266 (message):
  Error generating file
  build/CMakeFiles/cuda_compile.dir/tests/cpp/predictor/./cuda_compile_generated_test_gpu_predictor.cu.o


CMakeFiles/testxgboost.dir/build.make:1622: recipe for target 'CMakeFiles/cuda_compile.dir/tests/cpp/predictor/cuda_compile_generated_test_gpu_predictor.cu.o' failed
make[2]: *** [CMakeFiles/cuda_compile.dir/tests/cpp/predictor/cuda_compile_generated_test_gpu_predictor.cu.o] Error 1
CMakeFiles/Makefile2:150: recipe for target 'CMakeFiles/testxgboost.dir/all' failed
make[1]: *** [CMakeFiles/testxgboost.dir/all] Error 2
Makefile:94: recipe for target 'all' failed
make: *** [all] Error 2
```

## What have you tried?
Showing a patch that helped me resolve this case.
```
$ git diff include/xgboost/base.h
diff --git a/include/xgboost/base.h b/include/xgboost/base.h
index 24ff4d3..d6de038 100644
--- a/include/xgboost/base.h
+++ b/include/xgboost/base.h
@@ -6,6 +6,7 @@
 #ifndef XGBOOST_BASE_H_
 #define XGBOOST_BASE_H_

+#include <iostream>
 #include <dmlc/base.h>
 #include <dmlc/omp.h>
 #include <cmath>
```

--- a/include/xgboost/base.h
+++ b/include/xgboost/base.h
@@ -9,6 +9,7 @@
 #include <dmlc/base.h>
 #include <dmlc/omp.h>
 #include <cmath>
+#include <iostream>
 
 /*!
  * \brief string flag for R library, to leave hooks when needed.

--- a/tests/cpp/predictor/test_gpu_predictor.cu
+++ b/tests/cpp/predictor/test_gpu_predictor.cu
@@ -15,6 +15,9 @@ TEST(gpu_predictor, Test) {
   std::unique_ptr<Predictor> cpu_predictor =
       std::unique_ptr<Predictor>(Predictor::Create("cpu_predictor"));
 
+  gpu_predictor->Init({}, {});
+  cpu_predictor->Init({}, {});
+
   std::vector<std::unique_ptr<RegTree>> trees;
   trees.push_back(std::unique_ptr<RegTree>(new RegTree()));
   trees.back()->InitModel();

--- a/tests/cpp/tree/test_gpu_hist.cu
+++ b/tests/cpp/tree/test_gpu_hist.cu
@@ -22,9 +22,10 @@ TEST(gpu_hist_experimental, TestSparseShard) {
   hmat.Init(dmat.get(), max_bins);
   gmat.cut = &hmat;
   gmat.Init(dmat.get());
-  ncclComm_t comm;
+  TrainParam p;
+  p.max_depth = 6;
   DeviceShard shard(0, 0, gmat, 0, rows, hmat.row_ptr.back(),
-                    TrainParam());
+                    p);
 
   ASSERT_LT(shard.row_stride, columns);
 
@@ -56,9 +57,10 @@ TEST(gpu_hist_experimental, TestDenseShard) {
   hmat.Init(dmat.get(), max_bins);
   gmat.cut = &hmat;
   gmat.Init(dmat.get());
-  ncclComm_t comm;
+  TrainParam p;
+  p.max_depth = 6;
   DeviceShard shard(0, 0, gmat, 0, rows, hmat.row_ptr.back(),
-                    TrainParam());
+                    p);
 
   ASSERT_EQ(shard.row_stride, columns);
 

